packages/terra-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -372,23 +230,23 @@ function WorklistDataGrid(props) {
     }
   }, [intl, rows]);
 
-  // -------------------------------------
-
-  const isRowSelected = () => (
-    rows.find(row => row.isSelected === true)
-  );
-
-  const handleClearRowSelection = () => {
-    multiSelectRange.current = {}; // Clear the information used for selecting multiple rows.
-    if (isRowSelected()) {
-      // Esc (while in row selection mode and rows are selected): Clear selection
-      if (onClearSelectedRows) {
-        onClearSelectedRows();
+  const handleClearSelection = useCallback(() => {
+    if (!hasSelectableRows) {
+      if (onClearSelectedCells) {
+        onClearSelectedCells();
+      }
+    } else {
+      multiSelectRange.current = {}; // Clear the information used for selecting multiple rows.
+      if (rows.find(row => row.isSelected)) {
+        // Esc (while in row selection mode and rows are selected): Clear selection
+        if (onClearSelectedRows) {
```

Comment:
If this method is not defined, is there any need to check for selected rows?  It seems like moving the check up would make the logic in that scenario more efficient.